vizro-core/src/vizro/models/_navigation/_accordion.py

```diff
@@ -40,33 +42,20 @@ def _create_accordion_buttons(self, accordion_pages):
                 href=page["relative_path"],
             )
             for page in dash.page_registry.values()
-            for accordion_page in accordion_pages
-            if accordion_page == page["module"] and page["module"] != MODULE_PAGE_404
+            if page["name"] in pages
```

Comment:
Since you moved dash registry pages into the validator, does it mean we can use `self.pages` here?